### PR TITLE
Changes based on molecular mechanism search changes

### PR DESCRIPTION
**Pull request by olaaustine**
Requires [API-PR 42](https://github.com/EBI-G2P/gene2phenotype_api/pull/42) to be merged first

### CODE REVIEW

File: src/views/LocusGeneDiseaseView.vue [hunk 1]
```diff
@@ -410,20 +410,15 @@ export default {
...
             <td class="w-75">
-              <p
-                v-if="
-                  locusGeneDiseaseData.molecular_mechanism &&
-                  locusGeneDiseaseData.molecular_mechanism.length > 0
-                "
-              >
-                {{ locusGeneDiseaseData.molecular_mechanism[0].mechanism }}
+              <p v-if="locusGeneDiseaseData.molecular_mechanism">
+                {{ locusGeneDiseaseData.molecular_mechanism.mechanism }}
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:414]: Pls apply similar changes for mechanism categorisation and categorisation support

File: src/views/SearchPageView.vue [hunk 2]
```diff
@@ -141,6 +141,7 @@ export default {
               <th>Gene</th>
               <th>Disease</th>
               <th>Allelic Requirement</th>
+              <th>Molecular Mechanism</th>
```
Comment (seeta-ramaraju) [inline src/views/SearchPageView.vue:144]: In all tables, we are displaying the header as "Mechanism" so for consistency, better to keep display it as "Mechanism" for now

File: src/views/LocusGeneDiseaseView.vue (hunk 1 above)
Comment (olaaustine) [reply src/views/LocusGeneDiseaseView.vue:414]: Done, also add changes to evidence

File: src/views/LocusGeneDiseaseView.vue [hunk 3]
```diff
@@ -459,10 +448,7 @@ export default {
           <tr
             v-if="
               locusGeneDiseaseData.molecular_mechanism &&
-              locusGeneDiseaseData.molecular_mechanism.length > 0 &&
-              locusGeneDiseaseData.molecular_mechanism[0].evidence &&
-              Object.keys(locusGeneDiseaseData.molecular_mechanism[0].evidence)
-                .length > 0
+              locusGeneDiseaseData.molecular_mechanism.evidence
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:451]:
```suggestion
              locusGeneDiseaseData.molecular_mechanism?.evidence &&
              Object.keys(locusGeneDiseaseData.molecular_mechanism.evidence)
                .length > 0
```

File: src/views/LocusGeneDiseaseView.vue [hunk 4]
```diff
@@ -459,10 +448,7 @@ export default {
           <tr
             v-if="
               locusGeneDiseaseData.molecular_mechanism &&
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:450]: once the suggestion on line 451 "locusGeneDiseaseData.molecular_mechanism?.evidence" is added, this condition wont be required, please delete this condition

File: src/views/LocusGeneDiseaseView.vue [hunk 5]
```diff
@@ -434,22 +429,16 @@ export default {
...
             </td>
             <td class="w-75">
-              <p
-                v-if="
-                  locusGeneDiseaseData.molecular_mechanism &&
-                  locusGeneDiseaseData.molecular_mechanism.length > 0
-                "
-              >
-                {{ locusGeneDiseaseData.molecular_mechanism[0].synopsis }}
+              <p v-if="locusGeneDiseaseData.molecular_mechanism">
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:432]:
```suggestion
              <p v-if="locusGeneDiseaseData.molecular_mechanism?.synopsis">
```

File: src/views/LocusGeneDiseaseView.vue [hunk 6]
```diff
@@ -410,20 +410,15 @@ export default {
...
             </td>
             <td class="w-75">
-              <p
-                v-if="
-                  locusGeneDiseaseData.molecular_mechanism &&
-                  locusGeneDiseaseData.molecular_mechanism.length > 0
-                "
-              >
-                {{ locusGeneDiseaseData.molecular_mechanism[0].mechanism }}
+              <p v-if="locusGeneDiseaseData.molecular_mechanism">
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:413]:
```suggestion
              <p v-if="locusGeneDiseaseData.molecular_mechanism?.mechanism">
```

File: src/views/LocusGeneDiseaseView.vue [hunk 7]
```diff
@@ -410,20 +410,15 @@ export default {
...
+                {{ locusGeneDiseaseData.molecular_mechanism.mechanism }}
                 <span
                   v-if="
-                    locusGeneDiseaseData.molecular_mechanism[0].support ===
+                    locusGeneDiseaseData.molecular_mechanism.support ===
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:417]:
```suggestion
                    locusGeneDiseaseData.molecular_mechanism?.support ===
```

File: src/views/LocusGeneDiseaseView.vue [hunk 8]
```diff
@@ -434,22 +429,16 @@ export default {
...
-                  locusGeneDiseaseData.molecular_mechanism.length > 0
-                "
-              >
-                {{ locusGeneDiseaseData.molecular_mechanism[0].synopsis }}
+              <p v-if="locusGeneDiseaseData.molecular_mechanism">
+                {{ locusGeneDiseaseData.molecular_mechanism.synopsis }}
                 <span
                   v-if="
-                    locusGeneDiseaseData.molecular_mechanism[0]
+                    locusGeneDiseaseData.molecular_mechanism
```
Comment (seeta-ramaraju) [inline src/views/LocusGeneDiseaseView.vue:436]:
```suggestion
                    locusGeneDiseaseData.molecular_mechanism?
```

File: src/views/LocusGeneDiseaseView.vue (hunk 5 above)
Comment (seeta-ramaraju) [reply src/views/LocusGeneDiseaseView.vue:432]: checking the nested field "synopsis" is required.
Reason:
For scenario where there is no categorisation(synopsis) value, ideally we want the v-if condition to fail and go to v-else condition and display "Not Available" text.

**Comment by dglemos** [comment]
The mechanism evidence is not displayed on the record page.

**Review by seeta-ramaraju — APPROVED** [review]
LGTM

**Pull request merged**